lib/fog/aws/models/rds/server.rb

```diff
@@ -104,16 +104,22 @@ def modify(immediately, options)
         end
 
         def save
-          requires :engine
-          requires :allocated_storage
-          requires :master_username
-          requires :password
-
-          self.flavor_id ||= 'db.m1.small'
-
-          data = service.create_db_instance(id, attributes_to_params)
-          merge_attributes(data.body['CreateDBInstanceResult']['DBInstance'])
-          true
+          if source_snapshot_id
```

Comment:
Calling it restore is a probably a bit of a gaff on amazon's part, as it creates an entirely new server. There is no true "restore existing" functionality, you are just expected to create replacements.
 
The behavior of this `save` method therefore is consistent with `Volume#save` / `Volume.create` (optionally taking a snapshot_id).

see: http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_RestoreDBInstanceFromDBSnapshot.html